internal/pkg/service/appsproxy/proxy/server_test.go

```diff
@@ -0,0 +1,623 @@
+package proxy_test
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/umisama/go-regexpcache"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/config"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/dataapps/api"
+	proxyDependencies "github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/dependencies"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/proxy"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/proxy/testutil"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/dependencies"
+	"github.com/keboola/keboola-as-code/internal/pkg/telemetry"
+)
+
+func TestAppProxyHandler(t *testing.T) {
+	t.Parallel()
+
+	// Start app
+	appServer := testutil.StartAppServer(t)
+	defer appServer.Close()
+
+	// Start api
+	appsAPI := testutil.StartDataAppsAPI(t)
+	defer appsAPI.Close()
+
+	// Configure proxy
+	cfg := config.New()
+	cfg.API.PublicURL, _ = url.Parse("https://hub.keboola.local")
+	cfg.SandboxesAPI.URL = appsAPI.URL
+
+	// Create dependencies
+	d, mocked := proxyDependencies.NewMockedServiceScope(t, cfg, dependencies.WithRealHTTPClient())
+
+	// Register apps
+	appURL := testutil.AddAppDNSRecord(t, appServer, mocked.TestDNSServer())
+	appsAPI.Register([]api.AppConfig{
+		{
+			ID:             "123",
+			Name:           "public",
+			UpstreamAppURL: appURL.String(),
+			AuthRules: []api.Rule{
+				{
+					Type:         api.RulePathPrefix,
+					Value:        "/",
+					AuthRequired: pointer(false),
+				},
+			},
+		},
+	})
+
+	// Create proxy handler
+	handler := proxy.NewHandler(d)
+
+	// Get robots.txt
+	rec := httptest.NewRecorder()
+	req := httptest.NewRequest("GET", "https://hub.keboola.local/robots.txt", nil)
+	handler.ServeHTTP(rec, req)
+	require.Equal(t, http.StatusOK, rec.Code)
+	assert.Contains(t, rec.Body.String(), "Disallow: /")
```

Comment:
It is response for the `hub` root site, ... there is no `<appID>.hub` in the URL.

Reply:
So it is expected when you contact the `hub` without application that it returns OK and `Disallow: /` in body if I understand it right. I could not find such use case in RFC.

Reply:
Such details are not covered by the RFC.

`robots.txt` serves to prevent various web robots from scanning our proxy, because it would unnecessarily generate logs and warnings in the logs, I don't see any problem there (cc @jachym-tousek-keboola for check).

Reply:
For context: I noticed that `OAuth2Proxy` does this, and I realized that it is missing in our top-level handler.

Reply:
Can we also mention such technical detail in some document to always make sure that these handlers are working correcntly ? Only if it is possible :) . I know how `robots.txt` works but never made it in production so maybe I was not doing it correctly.

Reply:
:man_shrugging: it's a nice to have thing, if it wasn't there we would solve it probably as soon as we saw strange logs in DataDog.

It seems like common knowledge to me, just like we don't document what status code 200 means.

Reply:
Ok then I resolve this comment.